src/color-picker/color-picker.js

```diff
@@ -27,14 +30,22 @@ const Self = class ColorPicker extends ColorElement {
 		super.connectedCallback?.();
 		this._el.sliders.addEventListener("input", this);
 		this._el.swatch.addEventListener("input", this);
+		this._el.space_picker.addEventListener("spacechange", this);
```

Comment:
Wait, why does it fire spacechange and not input events? Does it do both? What's the difference? I thought spacechange was only for channel-picker, to distinguish what changed (though even there we may want to repurpose [`inputType`](https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/inputType) for that)

Reply:
> Does it do both? What's the difference?

It does. When the `<space-picker>` value is set programmatically, we can't use the `input` event to handle it. In this case, we indeed don't need it, so I switched to listening to `input` events on the slot, as you suggested. 

>(though even there we may want to repurpose [inputType](https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/inputType) for that)

I tried dispatching the `input` event using the [`InputEvent` constructor](https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/InputEvent) and passing the `inputType` option with the value of `spaceChange` (in the `<space-picker>` `input` event handler). Still, it seems it wasn't passed along. IDK why. 🤷‍♂️